src/main/java/Parser.java

```diff
@@ -0,0 +1,250 @@
+            }
+        } else {
+            throw new DukeException("Please enter a valid deadline");
+        }
+    }
+
+    public static void handleEventInput(String input, TaskList taskList) throws DukeException {
+        if (!input.substring(5).trim().isEmpty() //to make sure event is not empty
+                && input.substring(5).trim().contains("/at") //to make sure event contains at
+                && !input.substring(5).trim().startsWith("/at") //to make sure event description is not empty
+                && !input.substring(5).trim().endsWith("/at")) { //to make sure event contains a time/date
+            String descriptionAndTime = input.substring(5);
+            String description = descriptionAndTime.split("/at ")[0];
+            String at = descriptionAndTime.split("/at ")[1].trim();
+            try {
+                if (at.contains(" ")) { //user gave a time input
+                    String dateString = at.split(" ")[0].trim();
+                    String timeString = at.split(" ")[1].trim();
+                    dateString = formatDate(dateString);
+                    LocalDate d2 = LocalDate.parse(dateString);
+                    if (timeString.length() == 4) {
+                        try {
+                            int time = Integer.parseInt(timeString); //convert string to integer
+                            if (time >= 0000 && time <= 2359) {
+                               taskList.setEvent(description, d2, timeString);
+                            } else {
+                                throw new DukeException("Please enter a valid time between 0000 and 2359");
+                            }
+                        } catch (NumberFormatException nfe) {
+                            throw new DukeException("Please input the time in the right format (eg. 1800)");
+                        }
+                    } else {
+                        taskList.setEvent(description, d2);
+                    }
+                } else { //user didn't give a time input
+                    at = formatDate(at);
+                    LocalDate d2 = LocalDate.parse(at);
+                    taskList.setEvent(description, d2);
+                }
+            } catch (DateTimeException dte) {
+                throw new DukeException("Please enter your date and time in the format yyyy-mm-dd hhmm (eg. 2020-08-23 1800)");
+            }
+        } else {
+            throw new DukeException("Please enter a valid event");
+        }
+    }
+
+    public static TaskList handleFindInput(String input, TaskList taskList) throws DukeException {
+        if (!input.substring(4).trim().isEmpty()) { //to make sure to do task is not empty
+           return taskList.find(input.substring(4).trim());
+        } else {
+            throw new DukeException("Please enter a valid search item");
+        }
+    }
+
+    public static TaskList addTaskFromFile(String taskString, TaskList list) {
```

Comment:
Your variable name used to refer to the TaskList varies by using list here and taskList in other places, would it be better to use one name throughout to maintain consistency and reduce confusion? e.g. using name taskList throughout to refer to the variable TaskList